src/linux_io/remove_duplicate.cpp

```diff
@@ -0,0 +1,95 @@
+#include<set>
+#include<unistd.h>
+#include<string.h>
+#include<string>
+#include<iostream>
+#include<sys/stat.h>
+#include<fcntl.h>
+#include<stdio.h>
+#include<vector>
+#include<sstream>
+
+using namespace std;
+
+//要读取的文件
+const string file_name= "file.txt";
+const string write_name = "output.txt";
+//每次读200Bytes
+const size_t BUFFER_SIZE = 200;
+
+//将文件以每次200Byte读入vector
+vector<string> read_file(const string& file_name){
+	//打开文件,以只读权限打开	
+	int fd = open(file_name.c_str(), O_RDONLY);
+	if(fd == -1){
+		cout << "open file " << file_name << " failed" << endl;
+		exit(1);
+	}
+	//读取文件
+	//length为200或小于200的字符串
+	vector<string> data;
+	char buf[BUFFER_SIZE];
+	memset(buf,0,sizeof(buf));
+	ssize_t ret = 0;
+	while(ret = read(fd, buf, BUFFER_SIZE) != 0){
```

Comment:
![image](https://user-images.githubusercontent.com/1818023/69557534-b8c1fa80-0fe1-11ea-80fe-52f5e8df2061.png)

https://unix.stackexchange.com/questions/253349/eintr-is-there-a-rationale-behind-it